src/org/omegat/gui/glossary/IGlossaryRenderer.java

```diff
@@ -25,14 +25,133 @@
+
+    interface IRenderTarget<T> {
+        void append(String str);
+
+        void append(String str, AttributeSet attr);
+
+        T get();
+    }
+
+    static class DocTarget implements IRenderTarget<Void> {
+        DocTarget(StyledDocument doc) {
+            this.doc = doc;
+        }
+
+        private final StyledDocument doc;
+
+        @Override
+        public void append(String str) {
+            append(str, null);
+        }
+
+        @Override
+        public void append(String str, AttributeSet attr) {
+            try {
+                doc.insertString(doc.getLength(), str, attr);
+            } catch (BadLocationException e) {
+                // Should never happen since we only insert at end
+                Logger.getLogger(DefaultGlossaryRenderer.class.getName()).log(Level.SEVERE,
+                        e.getLocalizedMessage(), e);
+            }
+        }
+
+        @Override
+        public Void get() {
+            return null;
+        }
+    }
+
+    static class HtmlTarget implements IRenderTarget<String> {
+
+        private final StringBuilder buf = new StringBuilder();
+
+        @Override
+        public void append(String str) {
+            append(str, null);
+        }
+
+        @Override
+        public void append(String str, AttributeSet attr) {
+            if (attr != null) {
+                if (StyleConstants.isBold(attr)) {
+                    buf.append("<b>");
+                }
+                if (StyleConstants.isItalic(attr)) {
+                    buf.append("<i>");
+                }
+                Color attrColor = StyleConstants.getForeground(attr);
+                if (attrColor != Color.black) {
+                    String colorString = String.format("%02x%02x%02x",
+                            attrColor.getRed(), attrColor.getGreen(), attrColor.getBlue());
+                    buf.append("<font color=#" + colorString + ">");
+                }
+            }
+            buf.append(str);
+            if (attr != null) {
+                Color attrColor = StyleConstants.getForeground(attr);
+                if (attrColor != Color.black) {
+                    buf.append("</font>");
+                }
+                if (StyleConstants.isItalic(attr)) {
+                    buf.append("</i>");
+                }
+                if (StyleConstants.isBold(attr)) {
+                    buf.append("</b>");
+                }
+            }
+        }
+
+        @Override
+        public String get() {
+            return "<html><p>" + buf.toString().replace("\n", "<br>") + "</p></html>";
+        }
+    }
+
+    // --------------- Identification methods -----------------
+
+    /** Name to be displayed in the drop box. Can be language-dependant **/
+    String getName();
+
+    /** String to be stored in config file. Must be language-independant, and unique **/
+    String getId();
+
+    // --------------- Rendering methods -----------------
+
+    boolean wantsMerge();
```

Comment:
Sorry, I had forgotten to commit some piece of code. Have a look now at GlossarySearcher.java, line 285.

Reply:
Thanks. I understand now.

But merging/no-merging seems like a concern that is not related to the glossary renderer, since the only place it takes effect is *outside* of the rendering.

On the other hand I don't like adding a million toggles for every little thing. In fact I wonder if I said in a previous discussion that I would *not* want merging/no-merging to be a separate control. If I said that, it was probably under the (mistaken) assumption that merging takes place at render time rather than before it.

Seeing the result, however, I think adding "merge" and "no-merge" variants for every renderer is strictly worse than just having a separate toggle.

Reply:
I've moved this flag to a separate preference.

Reply:
I totally agree with this... but precisely, the problem is that _you_ said explicitly the contrary one year ago:
https://sourceforge.net/p/omegat/mailman/message/36777299/

In one of your messages you said that this feature was "orthogonal" to glossary layout: now I think that you understand that when two features are orthogonal, joining them in one means that the number of values in drop box is the scalar product of number of possibilities for both options!

The only reason why I had implemented like this is precisely the message I am quoting right now. But at the same time I had implemented in DGT a version with checkbox, exactly as what you are doing now. I always work like that, I always implement first in DGT and I advice to people, including you, to test the features in DGT - which does not mean that you have to use it after that. If you had tested the feature, instead of only speaking theoritically, you would have understood immediately that the checkbox did something totally different than glossary layout!